modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -128,5 +132,51 @@ object ConstructionService {
           }
         }
 
+    override def getPayloads(
+      operations: NonEmptyList[Operation],
+      maybeMetadata: Option[TransactionMetadata]
+    ): EitherT[F, ConstructionError, PayloadsResult] =
+      for {
+        _ <- EitherT.cond(operations.size == 2, (), IncorrectNumberOfOperations)
+
+        (positiveOperation, transactionAmount) <- EitherT.fromOption(
+          operations.collectFirst {
+            case op if op.amount.exists(!_.value.isNegative) =>
+              op.amount
+                .flatMap(_.value.toTransactionAmount.toOption)
+                .map((op, _))
+          }.flatten,
+          MissingPositiveAmountOperation
+        )
+
+        metadata <- EitherT.fromOption(maybeMetadata, MissingMetadata)
+
+        accountId <- EitherT.fromOption(positiveOperation.account, MissingAccountIdentifier)
+
+        dagTransaction =
+          DAGTransaction(
+            source = metadata.source,
```

Comment:
shouldn't it be taken from the operation itself? operation with positive balance reference a second operation with negative balance. imo, account address should be taken from that operation.

Reply:
@marcinwadon To clarify:

1. The two operations are mirrors of each other with the same amount, one negative and one positive
2. The _source_ address should come from the negative operation

Is that right?

Reply:
I wouldn't say they are mirrors, but in general if there is a Transfer operation, one must be with positive balance (to add coins/tokens to the assigned account) and the second one with negative balance (to remove coins/tokens from the assigned account). Maybe in this case they are mirrors, but that would be a specific case rather than typical.

and yes, source address is the one which operation has a negative amount (spends coins/tokens) while destination address receives coins/tokens (hence positive amount)